### PR TITLE
Windows script fix

### PR DESCRIPTION
**Pull request by sake92**
This PR fixes most of the issues with using CBT on Windows:
- cache strings are now *platform independent* as far as they can, by using `pip(unixPath: String)` (path-independent-path, just converts slashes to backslashes)
- batch script is now working properly and is *usable*

Some minor issues left:
- improve Nailgun support, `java.net.SocketException: Connection reset`
- time function
- nailgun log files are impossible to write to, when ng-server is up (`.. >> %nailgun_out% 2>> %nailgun_err%`)
- `eval` can't be used with single-quotes since CMD treats them specially... You can use double-quotes but then you can't use `String` literals 😄

### CODE REVIEW

File: nailgun_launcher/src/cbt/EarlyDependencies.java [hunk 1]
```diff
@@ -154,4 +157,13 @@ public EarlyDependencies(
...
+  
+  /** Platform independent path.<br>
+   * Replaces slashes with backslashes, e.g. something/bla becomes something\bla on Windows
+   */
+  private String pip(String pathWithSlashes) {
+	  // when replacing the path with \ Java treats it like escape character,
+	  // that's why we need Matcher.quoteReplacement
+	  return pathWithSlashes.replaceAll("/", Matcher.quoteReplacement(File.separator));
+  }
 }
```
Comment (cvogt) [inline nailgun_launcher/src/cbt/EarlyDependencies.java]: This file is auto-generated ( see comment in top line ). Editing it by hand is not the way to go.

Comment (sake92) [reply nailgun_launcher/src/cbt/EarlyDependencies.java]: Oh, I didn't see that. :/ Guess I'll have to find another way..

Comment (sake92) [reply nailgun_launcher/src/cbt/EarlyDependencies.java]: How about [this solution](https://github.com/cvogt/cbt/pull/577/files#diff-1b1258815a38bfcdf7559850f41a3037R168)?  
Btw, when I do `cbt tools cbtEarlyDependencies` it generates the file in `cbt/nailgun_launcher` and not in `cbt/nailgun_launcher/src/cbt`.. Is that expected?

Comment (cvogt) [reply nailgun_launcher/src/cbt/EarlyDependencies.java]: looks like a bug

**Comment by cvogt** [comment]
great stuff :)

**Pull request merged**